addons/hw_drivers/tools/iot_object_info_controller.py

```diff
@@ -0,0 +1,157 @@
+            ssl_certificate = {
+                'get_notAfter': ssl_certificate.get_notAfter(),
+                'get_notBefore': ssl_certificate.get_notBefore(),
+                'get_subject': ssl_certificate.get_subject(),
+                'get_issuer': ssl_certificate.get_issuer(),
+                'get_version': ssl_certificate.get_version(),
+                'get_serialNumber': ssl_certificate.get_serial_number(),
+                'get_signature_algorithm': ssl_certificate.get_signature_algorithm(),
+            }
+        iot_info_system = {
+            'platform': {
+                'system': platform_system,
+                'unamge': uname(),
+            },
+            'os': {
+                'path': os.getcwd(),
+                'login': getuser(),
+                'pid': os.getpid(),
+                'time': os.times(),
+            },
+            'netifaces': {
+                'AF_INET': netifaces.AF_INET,
+                'AF_INET6': netifaces.AF_INET6,
+                'AF_LINK': netifaces.AF_LINK,
+                'interfaces': {
+                    interface_name: netifaces.ifaddresses(interface_name)
+                    for interface_name in netifaces.interfaces()
+                }
+            },
+            'datetime': datetime.now(timezone.utc).isoformat(),
+            'helper checks/getters': {
+                'check_certificate': helpers.check_certificate(),
+                'get_certificate': ssl_certificate,
+                'check_image': helpers.check_image() if is_linux else '(only for Linux IoT-box)',
+                'get_ip': helpers.get_ip(),
+                'get_mac_address': helpers.get_mac_address(),
+                'get_path_nginx': helpers.get_path_nginx() if is_windows else '(only for Windows IoT)',
+                'get_ssid': helpers.get_ssid() if is_linux else '(only for Linux IoT-box)',
+                'get_odoo_server_url': helpers.get_odoo_server_url(),
+                'get_token': helpers.get_token(),
+                'get_version': helpers.get_version(),
+                'get_wifi_essid': helpers.get_wifi_essid() if is_linux else '(only for Linux IoT-box)',
+            },
+            'python': {
+                'version': sys.version,
+                'argv': sys.argv,
+                'path': sys.path,
+                'pip': sorted([f"{i.key}=={i.version}" for i in working_set]),
+            },
+            'odoo (used by IoT)': {
+                'version': exp_version(),
+                'config': tools.config.options,
+            }
+        }
+        if is_linux:
+            iot_info_system.update(
+                {
+                    'raspberry pi':
+                    {
+                        'board model': helpers.read_file_first_line('/proc/device-tree/model'),
+                        'vcgencmd': cls._iot_info_linux_vcgencmd(),
+                        'git': {
+                        'commit hash': check_output(
```

Comment:
make a function

Reply:
not the same function actually, we use log in the other case